test/nuclear-component.spec.js

```diff
@@ -0,0 +1,137 @@
+    }
+  }
+
+  describe('mount flow', () => {
+    let fakeReactor
+    let evaluateCalls = 0
+    let observeCalls = 0
+    let unwatchCalls = 0
+    let container
+
+    let div
+
+    before(() => {
+      fakeReactor = {
+        evaluate(key) {
+          ++evaluateCalls
+          return key
+        },
+        observe() {
+          ++observeCalls
+          return () => {
+            ++unwatchCalls
+          }
+        },
+      }
+    })
+
+    after(() => {
+      evaluateCalls = observeCalls = unwatchCalls = 0
+    })
+
+    it('should provide bindings as props on mount', () => {
+      @provideReactor
+      @nuclearComponent({
+        foo: 'bar',
+        test: 'lol',
+      })
+      class Container extends Component {
+        render() {
+          return <Child {...this.props}/>
+        }
+      }
+
+      div = document.createElement('div')
+
+      container = React.render(<Container pass='through' reactor={fakeReactor}/>, div)
+      const child = TestUtils.findRenderedComponentWithType(container, Child)
+      container.props.reactor.should.equal(fakeReactor)
+      container.props.pass.should.equal('through')
+      evaluateCalls.should.equal(2)
+      observeCalls.should.equal(2)
+      unwatchCalls.should.equal(0)
+      child.props.foo.should.equal('bar')
+      child.props.test.should.equal('lol')
+
+      Container.displayName.should.equal('ReactorProvider(NuclearComponent(Container))')
+    })
+
+    it('should correctly unmount', () => {
+      React.unmountComponentAtNode(div)
+      evaluateCalls.should.equal(2)
+      observeCalls.should.equal(2)
+      unwatchCalls.should.equal(2)
+    })
+  })
+
+  it('should not throw if we do not send dataBindings', () => {
+    @provideReactor
+    @nuclearComponent
+    class Container extends Component {
+      render() {
+        return <Child {...this.props}/>
+      }
+    }
+
+    const fakeReactor = {
+      evaluate() {},
+      observe() {
+        return () => {}
+      },
+    }
+
+    should.doesNotThrow(() => {
+      TestUtils.renderIntoDocument(<Container reactor={fakeReactor}/>)
+    })
+  })
+
+  it('should provide the reactor as context', () => {
+    @nuclearComponent
+    class Child2 extends Component {
+      render() {
+        return <div/>
+      }
+    }
+
+    class Child1 extends Component {
+      render() {
+        return <Child2/>
+      }
+    }
+
+    @provideReactor
+    class App extends Component {
+      render() {
+        return <Child1/>
+      }
+    }
+
+    const fakeReactor = {
+      evaluate() {},
+      observe() {
+        return () => {}
+      },
+    }
+
+    const app = TestUtils.renderIntoDocument(<App reactor={fakeReactor} />)
+    app.props.reactor.should.equal(fakeReactor)
+    const child1 = TestUtils.findRenderedComponentWithType(app, Child1)
+    const child2 = TestUtils.findRenderedComponentWithType(app, Child2)
+    should.not.exist(child1.context.reactor)
+    child2.context.reactor.should.equal(fakeReactor)
```

Comment:
I'm going to rework the test quickly to hopefully show why right now nuclearComponent is not working as it should.

Receiving the bindings as props works, but having access to the reactor somehow is not working, neither prop or context for now (if we accept that passing the reactor to the actions is how we're going to proceed going forward https://github.com/optimizely/nuclear-js/pull/106/files#diff-4659120dea035bc7b2aa83b2380494d9R53).